python/GafferImageUI/ImageViewUI.py

```diff
@@ -526,31 +673,43 @@ def __updateLazily( self ) :
 	@GafferUI.BackgroundMethod()
 	def __updateInBackground( self, source ) :
 
-		with Gaffer.Context( Gaffer.Context.current() ) as c :
-			if type( source ) == imath.V2i:
-				c["colorInspector:source"] = imath.V2f( source ) + imath.V2f( 0.5 ) # Center of pixel
-				color = self.getPlug().node()["colorInspector"]["evaluator"]["pixelColor"].getValue()
-			elif type( source ) == imath.Box2i:
-				areaEval = self.getPlug().node()["colorInspector"]["evaluator"]["areaColor"]
-				c["colorInspector:source"] = GafferImage.BufferAlgo.intersection( source, areaEval.getInput().node()["in"].format().getDisplayWindow() )
-				color = areaEval.getValue()
-			else:
-				raise Exception( "ColorInspector source must be V2i or Box2i, not " + str( type( source ) ) )
+		colors = [ imath.Color4f( 0, 0, 0, 1 ) ] * 2
+		for i in range( 2 ):
+			if self.__colorValueWidgets[i].getVisible():
```

Comment:
We're on a background thread here, so I don't think we can query the state of the widget.